internal/rbd/controllerserver.go

```diff
@@ -378,6 +394,14 @@ func (cs *ControllerServer) repairExistingVolume(ctx context.Context, req *csi.C
 
 			return nil, err
 		}
+
+		// Set metadata on restart of provisioner pod when image exist
+		if setMetadata {
+			err = rbdVol.setImageMetadata(ctx, req.GetParameters())
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
setmetadata should be done outside as it's applicable for both volume clone and volume creation